src/ring_theory/dedekind_domain/finite_adele_ring.lean

```diff
@@ -0,0 +1,272 @@
+/-
+Copyright (c) 2023 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import ring_theory.dedekind_domain.adic_valuation
+import topology.algebra.uniform_ring
+
+
+/-!
+# The finite adèle ring of a Dedekind domain
+We define the ring of finite adèles of a Dedekind domain `R`.
+
+## Main definitions
+- `dedekind_domain.finite_integral_adeles` : product of `adic_completion_integers`, where `v`
+  runs over all maximal ideals of `R`.
+- `dedekind_domain.prod_adic_completions` : the product of `adic_completion`, where `v` runs over
+  all maximal ideals of `R`.
+- `dedekind_domain.finite_adele_ring` : The finite adèle ring of `R`, defined as the
+  restricted product `Π'_v K_v`.
+
+## Implementation notes
+We are only interested on Dedekind domains of Krull dimension 1 (i.e., not fields). If `R` is a
+field, its finite adèle ring is just defined to be the trivial ring.
+
+## References
+* [J.W.S. Cassels, A. Frölich, *Algebraic Number Theory*][cassels1967algebraic]
+
+## Tags
+finite adèle ring, dedekind domain
+-/
+
+noncomputable theory
+open function set is_dedekind_domain is_dedekind_domain.height_one_spectrum
+
+namespace dedekind_domain
+
+variables (R K : Type*) [comm_ring R] [is_domain R] [is_dedekind_domain R] [field K]
+  [algebra R K] [is_fraction_ring R K] (v : height_one_spectrum R)
+
+/-- The product of all `adic_completion_integers`, where `v` runs over the maximal ideals of `R`. -/
+@[derive [comm_ring, topological_space, inhabited]]
+def finite_integral_adeles := Π (v : height_one_spectrum R), v.adic_completion_integers K
```

Comment:
```suggestion
def finite_integral_adeles : Type sorry :=
Π (v : height_one_spectrum R), v.adic_completion_integers K
```
where you should either use `#check` to work out what the sorry is, or just replace it with `*` if it's a mess